File: Rust/tests/test_min_operations.rs
use leetcode_rust::solution::Solution;

#[test]
fn test_example_1() {
    let input0 = vec![3, 9, 7];
    let input1 = 5;
    let expected = 4;
    assert_eq!(Solution::min_operations(input0, input1), expected);
}

#[test]
fn test_example_2() {
    let input0 = vec![4, 1, 3];
    let input1 = 4;
    let expected = 0;
    assert_eq!(Solution::min_operations(input0, input1), expected);
}

#[test]
fn test_example_3() {
    let input0 = vec![3, 2];
    let input1 = 6;
    let expected = 5;
    assert_eq!(Solution::min_operations(input0, input1), expected);
}

File: C/test/test_minOperations.c

#ifdef TEST

#include "unity.h"

#include "ltypes.h"
#include "minOperations.h"

void setUp(void)
{
}

void tearDown(void)
{
}

void test_minOperations_example_1(void)
{
	int input0[] = {3, 9, 7};
	int input1 = 5;
	int expected = 4;
	TEST_ASSERT_EQUAL_INT(expected, minOperations(input0, ARRAY_SIZE(input0), input1));
}

void test_minOperations_example_2(void)
{
	int input0[] = {4, 1, 3};
	int input1 = 4;
	int expected = 0;
	TEST_ASSERT_EQUAL_INT(expected, minOperations(input0, ARRAY_SIZE(input0), input1));
}

void test_minOperations_example_3(void)
{
	int input0[] = {3, 2};
	int input1 = 6;
	int expected = 5;
	TEST_ASSERT_EQUAL_INT(expected, minOperations(input0, ARRAY_SIZE(input0), input1));
}

#endif   // TEST
